tasks/collector.py

```diff
@@ -36,6 +59,86 @@
 }
 
 
+class YAMLValidationError(Exception):
+    def __init__(self, message):
+        super().__init__(message)
+
+
+def find_matching_components(manifest, components_to_match: dict, present: bool) -> list:
+    """Given a manifest and dict of components to match, if present=True, return list of
+    components found, otherwise return list of components missing."""
+    res = []
+    for component_type, components in components_to_match.items():
+        for component in components:
+            found_component = False
+            components_matching_component_type = manifest.get(component_type)
+            if components_matching_component_type:
+                for module in components_matching_component_type:
+                    if module.get("gomod").find(component) != -1:
+                        found_component = True
+                        if not present:
+                            found_component = True
```

Comment:
The second `found_component = True` is useless, are you sure the line 78 is correct ?

Reply:
good catch, thanks.

Reply:
I have fixed the function, thanks